### PR TITLE
feature: [PAR-4168] add husky precommit checks

### PR DESCRIPTION
**Pull request by milesmcleod**
* adds package.json to manage pre-commit linting dependencies: `prettier`, `lint-staged`, `husky`, and `@prettier/plugin-php`
* sets up `lint-staged` to run pre-commit, which is essentially just `prettier --write`

results:
* any staged changes will be linted via prettier upon commit
  * things like formatting white-spacing, quote marks, newlines, comments etc. will be written to the staged changes automatically and then committed
  * things like "syntax error: the word `spaghetti` is randomly sitting at the end of this file ⛔️" will loudly prevent the commit

### CODE REVIEW

File: Service/Extend.php [hunk 1]
```diff
@@ -24,4 +24,4 @@ class Extend
     public const COLLECT_ALL_ORDERS = true;
 
     public const SHIPPING_PROTECTION_LABEL = 'Shipping Protection';
-}
\ No newline at end of file
+}
```
Comment (milesmcleod) [inline Service/Extend.php]: dunno what this change is all about - i did use this file to test some linting stuff out which all looked good; my vscode is set up to ensure newlines at the end of every file by default on save, and it's showing up in my local, but not in this diff 🤷

File: package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "magento-extend-integration",
+  "description": "this npm package is used to manage linting dependencies only",
+  "version": "1.0.0",
+  "repository": "git@github.com:helloextend/magento-extend-integration.git",
+  "author": "Miles McLeod <miles.mcleod@extend.com>",
```
Comment (jay-faust-extend) [inline package.json]: Didn't realize we were re-branding, you _have_ to be the spokesperson!

Comment (milesmcleod) [reply package.json]: Oh boy. Gotta remove this haha

File: .nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+v18.15.0
```
Comment (jay-faust-extend) [inline .nvmrc:1]: I'm ignorant to what this file does in node core as well, I guess it's just essentially the target node version?

Edit: Google says yes

**Review by jay-faust-extend — APPROVED** [review]

File: .nvmrc (hunk 3 above)
Comment (milesmcleod) [reply .nvmrc:1]: yeah, it just indicates which node version should be used in order to ensure package dependency compatibility + code compatibility. `nvm use` in a folder containing `.nvmrc` will cause nvm to switch to the version specified in the file.

**Review by billhefty — APPROVED** [review]
Might be some tweaks we make the the lint rules as we go on, but solid baseline. Thanks!

**Pull request merged**